stdlib/LinearAlgebra/test/cholesky.jl

```diff
@@ -496,4 +496,56 @@ end
     @test B.UL ≈ B32.UL
 end
 
+@testset "det and logdet" begin
+    A = [4083 3825 5876 2048 4470 5490;
+         3825 3575 5520 1920 4200 5140;
+         5876 5520 8427 2940 6410 7903;
+         2048 1920 2940 1008 2240 2740;
+         4470 4200 6410 2240 4875 6015;
+         5490 5140 7903 2740 6015 7370]
+    B = cholesky(A, Val(true), check=false)
+    @test det(B)  ==  0.0
+    @test det(B)  ≈  det(A) atol=eps()
+    @test logdet(B)  ==  -Inf
+ end
+
+@testset "cholesky factorization success, and failure errors" begin
+    for T in (Float32, Float64, ComplexF64)
+        A = T[  2 -1 0;
+            -1 2 -1;
+            0 -1 2 ]
+        @test isposdef(A)
+        @test Hermitian(A)==A
+        @test LinearAlgebra.issuccess(cholesky(A; check = false))
+        @test LinearAlgebra.issuccess(cholesky!(copy(A); check = false))
+        B = T[  2 -1 0;
+                -1 2 -1;
+                0 -1 -5 ]
+        C = T[  2 -1 1;
+                -1 2 -1;
+                0 -1 2 ]
+        @test Hermitian(B)==B
+        @test Hermitian(C)!=C
+        for M in (B, C)
+            @test !isposdef(M)
+            @test_throws PosDefException cholesky(M)
+            @test_throws PosDefException cholesky!(copy(M))
+            @test_throws PosDefException cholesky(M; check = true)
+            @test_throws PosDefException cholesky!(copy(M); check = true)
+            @test !LinearAlgebra.issuccess(cholesky(M; check = false))
+            @test !LinearAlgebra.issuccess(cholesky!(copy(M); check = false))
+            @test_throws RankDeficientException cholesky(M, Val(true))
+            @test_throws RankDeficientException cholesky(M, Val(true); check = true)
+        end
+    end
+    D = ComplexF64[
+        5 2+3im 5im;
+        2-3im 7 1+7im;
+        -5im 1-7im 12]
+    @test isposdef(D)
+    @test Hermitian(D)==D
+    @test LinearAlgebra.issuccess(cholesky(D; check = false))
+    @test LinearAlgebra.issuccess(cholesky!(copy(D); check = false))
+end
```

Comment:
Looking at https://coveralls.io/builds/41286154/source?filename=stdlib%2FLinearAlgebra%2Fsrc%2Fcholesky.jl, it seems to me that none of these tests cover previously uncovered lines of code. What was your motivation to add these?

Reply:
This is actually my first time writing tests, and although I had checked [https://coveralls.io/github/JuliaLang/julia](https://coveralls.io/github/JuliaLang/julia) but I was not very careful about avoiding redundancy. I think all of these lines had been already covered, though some might have been covered only a handful of times, so I'm going to delete this test set in my next commit, now.

Reply:
Writing tests is really useful, but it's not about quantity, but about quality, which means covering many times the exact same functionality doesn't add any benefit. Having few tests which cover different cases (like different input types, different errors thrown, different branches exercisd etc.) is what is needed. So, keep on going, but please have an eye on which parts are not already covered, and then address those. This is much appreciated, believe me!

Reply:
Thank you for the clarification, will do!